react/src/components/PayButton/PayButton.tsx

```diff
@@ -67,11 +72,33 @@ export const PayButton = (props: PayButtonProps): React.ReactElement => {
   } = Object.assign({}, PayButton.defaultProps, props);
 
   const [paymentId] = useState(!disablePaymentId ? generatePaymentId(8) : undefined);
+  const delay = (ms: number) => new Promise(resolve => setTimeout(resolve, ms));
 
-  const handleButtonClick = (): void => {
-    if (onOpen !== undefined) onOpen(amount, to, paymentId);
+  useEffect(() => {
+    priceRef.current = price;
+  }, [price]);
+
+  useEffect(() => {
+    cryptoAmountRef.current = cryptoAmount;
+  }, [cryptoAmount]);
+
+  const waitPrice = async (callback: Function) => {
+    while (true) {
```

Comment:
this` while (true) ` loop with `await delay(300) ` could potentially lead to performance issues as it can lock the events loop, would be a better thing to use js tools like setInterval  https://stackoverflow.com/questions/14840527/whiletrue-vs-setintervalfunction-0#:~:text=There%20is%20a%20large%20difference,and%20depends%20on%20the%20browser.
`

Reply:
Refactored